v3/src/components/axis/hooks/use-axis.ts

```diff
@@ -17,124 +17,239 @@ export interface IUseAxis {
+        select(axisElt)
+          .attr("transform", initialTransform)
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore types are incompatible
+          .call(axis(scale).tickSizeOuter(0))
+          // Remove everything but the path the forms the axis line
+          .selectAll('g').remove()
+        // select(axisElt).selectAll('line').remove()
+        select(axisElt).append('g')
+          .transition().duration(duration)
+          .attr('transform', `translate(${axisIsVertical ? 0 : bandWidth / 2}, ` +
+            `${axisIsVertical ? bandWidth / 2 : 0})`)
+          .call(axis(scale).tickSizeInner(-tickLength))
+          .selectAll('.domain').remove()
+        let translation = '', rotation = '', textAnchor = 'none'
+        switch (axisPlace) {
+          case 'left':
+            switch (centerCategoryLabels) {
+              case true:
+                switch (collision) {
+                  case true:
+                    translation = `translate(0, ${-bandWidth / 2})`
+                    textAnchor = 'end'
+                    break
+                  case false:
+                    translation = `translate(${-textHeight / 2}, ${-bandWidth / 2})`
+                    rotation = `rotate(-90)`
+                    textAnchor = 'middle'
+                    break
+                }
+                break
+              case false:
+                switch (collision) {
+                  case true:
+                    translation = `translate(0, ${-textHeight / 2})`
+                    textAnchor = 'end'
+                    break
+                  case false:
+                    translation = `translate(${-textHeight / 2}, 0)`
+                    rotation = `rotate(-90)`
+                    textAnchor = 'start'
+                    break
+                }
+                break
+            }
+            break
+          case 'bottom':
+            switch (centerCategoryLabels) {
+              case true:
+                switch (collision) {
+                  case true:
+                    translation = `translate(${-bandWidth / 2 - textHeight / 2}, ${textHeight / 3})`
+                    rotation = `rotate(-90)`
+                    textAnchor = 'end'
+                    break
+                  case false:
+                    translation = `translate(${-bandWidth / 2}, 0)`
+                    textAnchor = 'middle'
+                    break
+                }
+                break
+              case false:
+                switch (collision) {
+                  case true:
+                    translation = `translate(${-bandWidth}, ${textHeight / 3})`
+                    rotation = `rotate(-90)`
+                    textAnchor = 'end'
+                    break
+                  case false:
+                    translation = `translate(${-bandWidth}, ${textHeight / 3})`
+                    textAnchor = 'start'
+                    break
+                }
```

Comment:
Since the translations are the same, this could be:
```suggestion
                translation = `translate(${-bandWidth}, ${textHeight / 3})`
                switch (collision) {
                  case true:
                    rotation = `rotate(-90)`
                    textAnchor = 'end'
                    break
                  case false:
                    textAnchor = 'start'
                    break
                }
```